editoast/src/views/stdcm.rs

```diff
@@ -0,0 +1,216 @@
+    let waypoints = parse_pathfinding_payload_waypoints(&data.steps, &track_map);
+    Ok(waypoints
+        .unwrap()
+        .iter()
+        .zip(data.steps.iter())
+        .map(|(wp, StepPayload { duration, .. })| STDCMCoreStep {
+            stop_duration: *duration,
+            stop: *duration > 0 as f64,
+            waypoints: wp.to_vec(),
+        })
+        .collect())
+}
+
+fn make_route_occupancies(
+    db_pool: Data<DbPool>,
+    timetable_id: i64,
+) -> Result<Vec<STDCMCoreRouteOccupancy>> {
+    let simulations = simulation_outputs_from_timetable_id(timetable_id, db_pool)?;
+    Ok(simulations
+        .iter()
+        .flat_map(|simulation| {
+            let sim = simulation
+                .eco_simulation
+                .as_ref()
+                .unwrap_or(&simulation.base_simulation);
+            sim.route_occupancies
+                .iter()
+                .map(|(route_id, occupancy)| STDCMCoreRouteOccupancy {
+                    id: route_id.to_string(),
+                    start_occupancy_time: occupancy.time_head_occupy
+                        + simulation.schedule_departure_time.unwrap(),
+                    end_occupancy_time: occupancy.time_tail_free
+                        + simulation.schedule_departure_time.unwrap(),
+                })
+        })
+        .collect())
+}
+
+async fn create_path_from_core_response(
+    db_pool: Data<DbPool>,
+    core_output: &STDCMCoreResponse,
+    data: &Json<STDCMRequestPayload>,
+) -> Result<STDCMPath> {
+    let core_path_response = &core_output.path;
+    let conn = &mut db_pool.get()?;
+    // not the same track ids as in api ??
+    // in api, every core_output.path —> route_paths[] —> track_sections[] —> track_id
+    // here, every core_output.path —> path_waypoints[] —> track_id
+    let track_map = core_path_response.fetch_track_map(data.infra_id, conn)?;
+    let op_map = core_path_response.fetch_op_map(data.infra_id, conn)?;
+    let pathfinding_from_response = Pathfinding::from_core_response(
+        data.steps.clone(),
+        core_path_response,
+        &track_map,
+        &op_map,
+    )?;
+    let pathfinding = PathfindingChangeset {
+        id: None,
+        infra_id: Some(data.infra_id),
+        ..pathfinding_from_response.into()
+    }
+    .create(db_pool)
+    .await?
+    .into();
+    Ok(pathfinding)
+}
+
+async fn create_simulation_from_core_response(
+    db_pool: Data<DbPool>,
+    core: &Data<CoreClient>,
+    data: &Json<STDCMRequestPayload>,
+    core_output: &STDCMCoreResponse,
+    path: &STDCMPath,
+) -> Result<SimulationReport> {
+    let schedule = TrainSchedule {
+        id: None,
+        departure_time: core_output.departure_time,
+        initial_speed: 0.0,
+        comfort: data.comfort.to_string(),
+        path_id: path.id,
+        rolling_stock_id: data.rolling_stock_id,
+        timetable_id: data.timetable_id,
+        ..Default::default()
+    };
+    let projection = Projection::new(&path.payload);
```

Comment:
We had a hard time understanding what `Projection` is (thanks @Khoyo 😁) but since we're at it, document it so that we don't forget (I definitely will).